src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,170 @@
 package core.basesyntax;
 
+//import jdk.internal.util.ArraysSupport;
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
+
 public class ArrayList<T> implements List<T> {
+
+
+    private static final int def_capacity = 10;
+    private static final Object[] empty_arraydata = {};
+
+
+    private T[] array = (T[]) new Object[def_capacity];
```

Comment:
The primary difference between arrays and generics when it comes to variance is that arrays are covariant while generics are invariant. Arrays allow for subtyping between related types, while generics are strict and do not allow for subtyping between related types. 
This means that an array of one type can be assigned to a variable of another type, while a generic type cannot.
An array can be assigned to a variable of type Object, but generics cannot.
Generics are checked at compile-time, and arrays are only at runtime.
As long as the very first rule of this Task was Don’t use class Objects, what possible could go wrong?
So there was a way to overcome this challenge:
To use  Java’s reflection facility with @SuppressWarnings("unchecked") and prey you made no mistake.